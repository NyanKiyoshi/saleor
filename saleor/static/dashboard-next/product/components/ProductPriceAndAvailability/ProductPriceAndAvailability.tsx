import Card, { CardContent } from "material-ui/Card";
import green from "material-ui/colors/green";
import red from "material-ui/colors/red";
import { FormControlLabel } from "material-ui/Form";
import { withStyles } from "material-ui/styles";
import Switch from "material-ui/Switch";
import Table, { TableBody, TableCell, TableRow } from "material-ui/Table";
import Typography from "material-ui/Typography";
import * as React from "react";

import ControlledSwitch from "../../../components/ControlledSwitch";
import i18n from "../../../i18n";

interface ProductPriceAndAvailabilityProps {
  product: {
    id: string;
    productType: {
      id: string;
    };
    name: string;
    description: string;
    category: {
      id: string;
    };
    price: {
      currency: string;
      amount: number;
      localized: string;
    };
    availableOn?: string;
    isPublished: boolean;
    attributes: string;
    updatedAt: string;
    availability: {
      available: boolean;
      onSale: boolean;
    };
    images: Array<{
      id: string;
      alt: string;
      url: string;
      order: number;
    }>;
  };
  onPublish(event: React.ChangeEvent<any>);
}

const decorate = withStyles(theme => ({
  greenText: {
    color: green[500]
  },
  leftCell: {
    paddingRight: theme.spacing.unit
  },
  redText: {
    color: red[500]
  },
  rightCell: {
    paddingLeft: theme.spacing.unit,
    textAlign: "right"
  }
}));
export const ProductPriceAndAvailability = decorate<
  ProductPriceAndAvailabilityProps
>(({ classes, product, onPublish }) => (
  <Card>
    <CardContent>
      <ControlledSwitch
        onChange={onPublish}
        uncheckedLabel={i18n.t("Draft")}
        label={i18n.t("Published")}
        checked={product.isPublished}
      />
      <Typography>
        {product.availability.available ? (
          <span className={classes.greenText}>{i18n.t("Available")}</span>
        ) : (
          <span className={classes.redText}>{i18n.t("Unavailable")}</span>
        )}
      </Typography>
    </CardContent>
    <Table>
      <TableBody>
        <TableRow>
          <TableCell className={classes.leftCell}>
            {i18n.t("Sale price")}
          </TableCell>
          <TableCell className={classes.rightCell}>
            {product.price.localized} - {product.price.localized}
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell className={classes.leftCell}>
            {i18n.t("Purchase cost")}
          </TableCell>
          <TableCell className={classes.rightCell}>
            {product.price.localized} - {product.price.localized}
          </TableCell>
        </TableRow>
        <TableRow>
          <TableCell className={classes.leftCell}>
            {i18n.t("Gross margin")}
          </TableCell>
          <TableCell className={classes.rightCell}>
            {product.price.localized} - {product.price.localized}
          </TableCell>
        </TableRow>
      </TableBody>
    </Table>
  </Card>
));
export default ProductPriceAndAvailability;
