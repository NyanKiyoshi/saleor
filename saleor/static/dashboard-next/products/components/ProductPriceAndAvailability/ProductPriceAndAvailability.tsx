import Card, { CardContent } from "material-ui/Card";
import green from "material-ui/colors/green";
import red from "material-ui/colors/red";
import { FormControlLabel } from "material-ui/Form";
import { withStyles } from "material-ui/styles";
import Switch from "material-ui/Switch";
import Table, { TableBody, TableCell, TableRow } from "material-ui/Table";
import Typography from "material-ui/Typography";
import * as React from "react";

import ControlledSwitch from "../../../components/ControlledSwitch";
import Skeleton from "../../../components/Skeleton";
import i18n from "../../../i18n";

interface ProductPriceAndAvailabilityProps {
  grossMargin?: Array<{
    start: number;
    stop: number;
  }>;
  salePrice?: {
    start: {
      net: {
        localized: string;
      };
    };
    stop: {
      net: {
        localized: string;
      };
    };
  };
  purchaseCost?: {
    start: {
      gross: {
        localized: string;
      };
    };
    stop: {
      gross: {
        localized: string;
      };
    };
  };
  isPublished?: boolean;
  isAvailable?: boolean;
  onPublish(event: React.ChangeEvent<any>);
}

const decorate = withStyles(theme => ({
  greenText: {
    color: green[500]
  },
  leftCell: {
    paddingRight: theme.spacing.unit
  },
  redText: {
    color: red[500]
  },
  rightCell: {
    paddingLeft: theme.spacing.unit,
    textAlign: "right"
  }
}));
export const ProductPriceAndAvailability = decorate<
  ProductPriceAndAvailabilityProps
>(
  ({
    classes,
    grossMargin,
    salePrice,
    purchaseCost,
    isPublished,
    isAvailable,
    onPublish
  }) => (
    <Card>
      <CardContent>
        <ControlledSwitch
          onChange={onPublish}
          uncheckedLabel={
            isPublished === undefined || isPublished === null
              ? " "
              : i18n.t("Draft")
          }
          label={i18n.t("Published")}
          checked={!(isPublished === undefined || isPublished === null)}
          disabled={isPublished === undefined || isPublished === null}
        />
        <Typography>
          {isAvailable === undefined || isAvailable === null ? (
            <Skeleton />
          ) : isAvailable ? (
            <span className={classes.greenText}>{i18n.t("Available")}</span>
          ) : (
            <span className={classes.redText}>{i18n.t("Unavailable")}</span>
          )}
        </Typography>
      </CardContent>
      <Table>
        <TableBody>
          <TableRow>
            <TableCell className={classes.leftCell}>
              {i18n.t("Sale price")}
            </TableCell>
            <TableCell className={classes.rightCell}>
              {salePrice ? (
                <span>
                  {salePrice.start.net.localized} -{" "}
                  {salePrice.stop.net.localized}
                </span>
              ) : (
                <Skeleton />
              )}
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell className={classes.leftCell}>
              {i18n.t("Purchase cost")}
            </TableCell>
            <TableCell className={classes.rightCell}>
              {purchaseCost ? (
                <span>
                  {purchaseCost.start.gross.localized} -{" "}
                  {purchaseCost.stop.gross.localized}
                </span>
              ) : (
                <Skeleton />
              )}
            </TableCell>
          </TableRow>
          <TableRow>
            <TableCell className={classes.leftCell}>
              {i18n.t("Gross margin")}
            </TableCell>
            <TableCell className={classes.rightCell}>
              {grossMargin ? (
                <span>
                  {grossMargin[0].start}% - {grossMargin[0].stop}%
                </span>
              ) : (
                <Skeleton />
              )}
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </Card>
  )
);
ProductPriceAndAvailability.displayName = "ProductPriceAndAvailability";
export default ProductPriceAndAvailability;
