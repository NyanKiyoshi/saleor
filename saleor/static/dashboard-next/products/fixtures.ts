export const product = {
  id: "UHJvZHVjdDoyOA==",
  name: "Stewart-Barrera",
  description:
    "Light light office style present life military. Enough executive coach cause own.\n\nCare season wrong fly. Example age between discuss assume.\n\nTheir third build might strong account. Worry enter room note level air.\n\nWeight director mother. Because find task with now enter tree management.\n\nSoon name recently ground green. There owner sport box them red information.",
  collections: {
    edges: [
      {
        node: {
          id: "collection-1",
          name: "Winter Collection"
        }
      },
      {
        node: {
          id: "collection-2",
          name: "Summer Collection"
        }
      }
    ]
  },
  price: {
    localized: "95,47 USD"
  },
  grossMargin: [
    {
      start: 56,
      stop: 98
    }
  ],
  purchaseCost: {
    start: {
      gross: {
        localized: "1,88 USD"
      }
    },
    stop: {
      gross: {
        localized: "48,10 USD"
      }
    }
  },
  priceRange: {
    start: {
      net: {
        localized: "55,14 USD"
      }
    },
    stop: {
      net: {
        localized: "88,36 USD"
      }
    }
  },
  isPublished: true,
  availability: {
    available: true
  },
  images: {
    edges: [
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY2",
          alt: "",
          order: 0,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-02_IS8e8b3.jpg"
        }
      },
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY3",
          alt: "",
          order: 1,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-02_ZIk8H8C.jpg"
        }
      },
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY4",
          alt: "",
          order: 2,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-03_i5b2KpM.jpg"
        }
      }
    ]
  },
  variants: {
    edges: [
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6Nw==",
          sku: "2-1337",
          name: "XXL",
          priceOverride: null,
          stockQuantity: 5
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6OA==",
          sku: "2-1338",
          name: "XL",
          priceOverride: null,
          stockQuantity: 0
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6OQ==",
          sku: "2-1339",
          name: "L",
          priceOverride: null,
          stockQuantity: 15
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTA=",
          sku: "2-1340",
          name: "M",
          priceOverride: null,
          stockQuantity: 33
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTE=",
          sku: "2-1341",
          name: "S",
          priceOverride: null,
          stockQuantity: 0
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTI=",
          sku: "2-1342",
          name: "XS",
          priceOverride: null,
          stockQuantity: 0
        }
      }
    ]
  },
  productType: {
    id: "UHJvZHVjdFR5cGU6Mw==",
    name: "Coffee"
  },
  url: "/pl/products/stewart-barrera-28/"
};
export const products = [
  {
    id: "UHJvZHVjdDox",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoy",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoz",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoa",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  }
];

export const variants = product.variants.edges.map(edge => edge.node);
export const grossMargin = product.grossMargin[0];
export const salePrice = {
  start: product.priceRange.start.net.localized,
  stop: product.priceRange.stop.net.localized
};
export const purchaseCost = {
  start: product.purchaseCost.start.gross.localized,
  stop: product.purchaseCost.stop.gross.localized
};
export const collections = product.collections.edges.map(edge => edge.node);
export const images = product.images.edges.map(edge => edge.node);
