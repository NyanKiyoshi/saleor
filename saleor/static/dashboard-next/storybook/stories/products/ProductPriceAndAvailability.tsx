import { storiesOf } from "@storybook/react";
import * as React from "react";

import ProductPriceAndAvailability from "../../../products/components/ProductPriceAndAvailability";

const grossMargin = {
  start: "30 $",
  stop: "40 $"
};
const salePrice = {
  start: "30 $",
  stop: "40 $"
};
const purchaseCost = {
  start: "30 $",
  stop: "40 $"
};
storiesOf("Products / ProductPriceAndAvailability", module)
  .add("when loading data", () => (
    <ProductPriceAndAvailability onPublish={() => {}} />
  ))
  .add("when product is available", () => (
    <ProductPriceAndAvailability
      onPublish={() => {}}
      grossMargin={grossMargin}
      salePrice={salePrice}
      purchaseCost={purchaseCost}
      isAvailable={true}
    />
  ))
  .add("when product is not available", () => (
    <ProductPriceAndAvailability
      onPublish={() => {}}
      grossMargin={grossMargin}
      salePrice={salePrice}
      purchaseCost={purchaseCost}
      isAvailable={false}
    />
  ));
