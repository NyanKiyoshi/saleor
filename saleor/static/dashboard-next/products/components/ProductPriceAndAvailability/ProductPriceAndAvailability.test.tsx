import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import { grossMargin, purchaseCost, salePrice } from "../../fixtures";
import ProductPriceAndAvailability from "./";

describe("<ProductPriceAndAvailability />", () => {
  it("renders when loading data", () => {
    const component = renderer.create(
      <ProductPriceAndAvailability onPublish={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when product is available", () => {
    const component = renderer.create(
      <ProductPriceAndAvailability
        onPublish={() => {}}
        grossMargin={grossMargin}
        salePrice={salePrice}
        purchaseCost={purchaseCost}
        isAvailable={true}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders product is not available", () => {
    const component = renderer.create(
      <ProductPriceAndAvailability
        onPublish={() => {}}
        grossMargin={grossMargin}
        salePrice={salePrice}
        purchaseCost={purchaseCost}
        isAvailable={true}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
