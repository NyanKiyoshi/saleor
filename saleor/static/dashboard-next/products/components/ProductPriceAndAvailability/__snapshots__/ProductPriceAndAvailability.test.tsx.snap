// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPriceAndAvailability /> renders product is not available 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="MuiCardContent-root-32"
  >
    <label
      className="MuiFormControlLabel-root-33 MuiFormControlLabel-disabled-34"
    >
      <span
        className="MuiSwitch-root-36"
      >
        <span
          className="MuiButtonBase-root-56 MuiButtonBase-disabled-57 MuiIconButton-root-50 MuiIconButton-disabled-54 MuiSwitchBase-root-45 MuiSwitchBase-default-47 MuiSwitch-default-40 MuiSwitchBase-disabled-49 MuiSwitch-disabled-44"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex="-1"
        >
          <span
            className="MuiIconButton-label-55"
          >
            <span
              className="MuiSwitch-icon-38"
            />
            <input
              checked={false}
              className="MuiSwitchBase-input-46"
              disabled={true}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
        </span>
        <span
          className="MuiSwitch-bar-37"
        />
      </span>
      <span
        className="MuiTypography-root-59 MuiTypography-body1-68 MuiFormControlLabel-label-35"
      >
         
      </span>
    </label>
    <p
      className="MuiTypography-root-59 MuiTypography-body1-68"
    >
      <span
        className="Component-greenText-1"
      >
        Available
      </span>
    </p>
  </div>
  <table
    className="MuiTable-root-84"
  >
    <tbody
      className="MuiTableBody-root-85"
    >
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Sale price
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            55,14 USD
             - 
            88,36 USD
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Purchase cost
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            1,88 USD
             - 
            48,10 USD
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Gross margin
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            56
            % - 
            98
            %
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductPriceAndAvailability /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="MuiCardContent-root-32"
  >
    <label
      className="MuiFormControlLabel-root-33 MuiFormControlLabel-disabled-34"
    >
      <span
        className="MuiSwitch-root-36"
      >
        <span
          className="MuiButtonBase-root-56 MuiButtonBase-disabled-57 MuiIconButton-root-50 MuiIconButton-disabled-54 MuiSwitchBase-root-45 MuiSwitchBase-default-47 MuiSwitch-default-40 MuiSwitchBase-disabled-49 MuiSwitch-disabled-44"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex="-1"
        >
          <span
            className="MuiIconButton-label-55"
          >
            <span
              className="MuiSwitch-icon-38"
            />
            <input
              checked={false}
              className="MuiSwitchBase-input-46"
              disabled={true}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
        </span>
        <span
          className="MuiSwitch-bar-37"
        />
      </span>
      <span
        className="MuiTypography-root-59 MuiTypography-body1-68 MuiFormControlLabel-label-35"
      >
         
      </span>
    </label>
    <p
      className="MuiTypography-root-59 MuiTypography-body1-68"
    >
      <span
        className="Component-skeleton-83"
        style={undefined}
      >
        ‌
      </span>
    </p>
  </div>
  <table
    className="MuiTable-root-84"
  >
    <tbody
      className="MuiTableBody-root-85"
    >
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Sale price
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span
            className="Component-skeleton-83"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Purchase cost
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span
            className="Component-skeleton-83"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Gross margin
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span
            className="Component-skeleton-83"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductPriceAndAvailability /> renders when product is available 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="MuiCardContent-root-32"
  >
    <label
      className="MuiFormControlLabel-root-33 MuiFormControlLabel-disabled-34"
    >
      <span
        className="MuiSwitch-root-36"
      >
        <span
          className="MuiButtonBase-root-56 MuiButtonBase-disabled-57 MuiIconButton-root-50 MuiIconButton-disabled-54 MuiSwitchBase-root-45 MuiSwitchBase-default-47 MuiSwitch-default-40 MuiSwitchBase-disabled-49 MuiSwitch-disabled-44"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex="-1"
        >
          <span
            className="MuiIconButton-label-55"
          >
            <span
              className="MuiSwitch-icon-38"
            />
            <input
              checked={false}
              className="MuiSwitchBase-input-46"
              disabled={true}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
        </span>
        <span
          className="MuiSwitch-bar-37"
        />
      </span>
      <span
        className="MuiTypography-root-59 MuiTypography-body1-68 MuiFormControlLabel-label-35"
      >
         
      </span>
    </label>
    <p
      className="MuiTypography-root-59 MuiTypography-body1-68"
    >
      <span
        className="Component-greenText-1"
      >
        Available
      </span>
    </p>
  </div>
  <table
    className="MuiTable-root-84"
  >
    <tbody
      className="MuiTableBody-root-85"
    >
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Sale price
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            55,14 USD
             - 
            88,36 USD
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Purchase cost
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            1,88 USD
             - 
            48,10 USD
          </span>
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-86"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-leftCell-2"
          scope={undefined}
        >
          Gross margin
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-91 MuiTableCell-body-93 Component-rightCell-4"
          scope={undefined}
        >
          <span>
            56
            % - 
            98
            %
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
